### PR TITLE
Second round of changes to the MCU docs

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

File: api/rs/slint/mcu.md [hunk 1]
```diff
@@ -1,25 +1,25 @@
 # Slint on Microcontrollers (MCU)
 
-This document explains how to use Slint to develop a UI on a MCU in a bare metal environment.
+The following scetions explain how to use Slint to develop a UI on a MCU in a bare metal environment.
```
Comment (ogoffart) [inline api/rs/slint/mcu.md]:
```suggestion
The following sections explain how to use Slint to develop a UI in a bare metal environment.
```

File: api/rs/slint/mcu.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 
-Covering these is out of scope for this document, but we recommend reading the [Rust Embedded Book](https://docs.rust-embedded.org/book/),
+We recommend reading the [Rust Embedded Book](https://docs.rust-embedded.org/book/),
 as well as the curated list of [Awesome Embedded Rust](https://github.com/rust-embedded/awesome-embedded-rust) for a wide range of different
 crates, tools and training materials. These resources should guide you through the initial setup and often come with "hello world" examples
 to get started with your device.
 
-In order to set a global allocator, required by Slint, you will need a **nightly** version of Rust. This is due to the fact that the support for using a custom global allocator in a bare metal
-environment with `#![no_std]` has not been stabilized yet (see [#51540](https://github.com/rust-lang/rust/issues/51540) or
+Slint requires a global memory allocator. That is available the nightly version of Rust, because the support for using a custom global
```
Comment (ogoffart) [inline api/rs/slint/mcu.md]:
```suggestion
Slint requires a global memory allocator. That is only possible with the nightly version of Rust, because the support for using a custom global
```

File: api/rs/slint/mcu.md [hunk 3]
```diff
@@ -38,28 +38,21 @@ features = ["compat-0.3.0", "unsafe-single-threaded", "libm"]
...
  * `unsafe-single-threaded`: Slint internally uses Rust's [`thread_local!`](https://doc.rust-lang.org/std/macro.thread_local.html) macro to store global data.
-   This feature is only available in the Rust Standard Library (std), which is not available in bare-metal environments. As a fallback, the `unsafe-single-threaded`
-   feature will change Slint to use unsafe static for storage. By setting this feature, you guarantee not to use Slint API from a thread other than the main thread,
-   or from interrupt handlers.
- * `libm`: The Rust Standard Library (std) provides traits and functions for floating point arithmetic. Without `std`, this feature enables the use of the
-   [libm](https://crates.io/crates/libm) crate to substitute this functionality.
+   This feature is only available in the Rust Standard Library (std), but is not available in bare metal environments. As a fallback, the `unsafe-single-threaded`
+   feature changes Slint to use unsafe static for storage. This way, you guarantee to use Slint API only from the main thread, or from interrupt handlers.
```
Comment (ogoffart) [inline api/rs/slint/mcu.md:49]:
```suggestion
   This macro is only available in the Rust Standard Library (std), not in bare metal environments. As a fallback, the `unsafe-single-threaded`
   feature changes Slint to use unsafe static for storage. By selecting this feature, you must ensure that you only use
   the Slint API from a single thread and not from interrupt handlers.
```

The sense of the sentence was changed: It is **not** possible to use slint from interrupt handlers

File: api/rs/slint/mcu.md [hunk 4]
```diff
@@ -225,24 +221,22 @@ loop {
 
 ### The Renderer
 
-In desktop and embedded environments, Slint typically uses operating system provided, often hardware-accelerated APIs to render the user interface.
+In desktop and embedded environments, Slint typically uses operating system provided APIs to render the user interface. These APIs are often hardware accelerated.
 In contrast, most MCUs don't have dedicated chips to render advanced graphics. Instead, the CPU is responsible for computing the colors of each
 pixels on the screen. This is called software rendering, and Slint provides a software renderer for this task.
```
Comment (ogoffart) [inline api/rs/slint/mcu.md:226]: I'm not sure it is so good to say that "most MCUs don't have dedicated chips to render advanced graphics", while many do. (eg, DMA2D or other DMA-like accelerator)
This could confuse people who would think they cannot use the software renderer to use DMA2D, for example

My suggestion:

```suggestion
In desktop and embedded environments, Slint typically uses operating system provided APIs to render the user interface using the GPU.
In contrast, most MCUs don't have GPUs. Instead, the all the rendering is done by sotfware on the CPU. This is called software rendering, and Slint provides a SoftwareRenderer for this task.
```

File: api/rs/slint/mcu.md [hunk 5]
```diff
@@ -379,17 +373,17 @@ loop {
...
 
-Hint: In our experience, using the synchronous `DrawTarget::fill_contiguous` function is slow. If
+Note: In our experience, using the synchronous `DrawTarget::fill_contiguous` function is slow. If
 your device is capable of using DMA, you may be able to achieve better performance by using
 two line buffers: One buffer to render into with the CPU, while the other buffer is transferred to
 the screen using DMA asynchronously.
 
-Hint: If you see wrong colors on your device, it may be necessary to invert the bits in the u16 pixel
+Note: If you see wrong colors on your screen, it may be necessary to invert the bits in the u16 pixel
 representation before sending them to the screen driver.
```
Comment (ogoffart) [inline api/rs/slint/mcu.md:383]:
```suggestion
```

If the user uses DrawTarget::fill_contiguous, this shouldn't be a problem at all.
otherwise it might just be a misconfigured driver or something. I believe most developer would figure that out by themselves.

File: api/rs/slint/mcu.md (hunk 1 above)
Comment (tronical) [reply api/rs/slint/mcu.md]: I fixed the typo, but I'm unsure about the removal of the MCU. The examples below refer to HAL crates that I think are clearly MCU material. How strong do you feel about it?

File: api/rs/slint/mcu.md (hunk 2 above)
Comment (tronical) [reply api/rs/slint/mcu.md]: I used "That is currently only available in the nightly version" to emphasise that this will change in the future (hopefully :-). Is that ok?

File: api/rs/slint/mcu.md (hunk 3 above)
Comment (tronical) [reply api/rs/slint/mcu.md]: I remove the redundant "available" (well spotted).

Well spotted about the change of meaning!

File: api/rs/slint/mcu.md (hunk 5 above)
Comment (tronical) [reply api/rs/slint/mcu.md:382]: Ok, if you think so. We can add it again if it becomes a FAQ.

File: api/rs/slint/mcu.md (hunk 1 above)
Comment (ogoffart) [reply api/rs/slint/mcu.md]: not strongly

File: api/rs/slint/mcu.md (hunk 2 above)
Comment (ogoffart) [reply api/rs/slint/mcu.md]: That's Ok,  although i liked "possible" more :-)

**Review by hunger — COMMENTED** [review]
Suggest rewording

File: api/rs/slint/mcu.md (hunk 2 above)
Comment (tronical) [reply api/rs/slint/mcu.md]: Ahh, you've convinced me. After all, they might decide to implement this using different API if they stabilise it, and then it's about the possibility of doing it, not about the availability of the concrete API that's in nightly right now. I'll change it :)

**Comment by tronical** [comment]
> Suggest rewording

That's very broad :-)

File: api/rs/slint/mcu.md [hunk 6]
```diff
@@ -1,29 +1,29 @@
...
+We recommend reading the [Rust Embedded Book](https://docs.rust-embedded.org/book/),
 as well as the curated list of [Awesome Embedded Rust](https://github.com/rust-embedded/awesome-embedded-rust) for a wide range of different
 crates, tools and training materials. These resources should guide you through the initial setup and often come with "hello world" examples
 to get started with your device.
 
-In order to set a global allocator, required by Slint, you will need a **nightly** version of Rust. This is due to the fact that the support for using a custom global allocator in a bare metal
-environment with `#![no_std]` has not been stabilized yet (see [#51540](https://github.com/rust-lang/rust/issues/51540) or
+Slint requires a global memory allocator. That is currently only available in the nightly version of Rust, because the support for using a custom global
+allocator in a bare metal environment with `#![no_std]` has not been stabilized yet (see [#51540](https://github.com/rust-lang/rust/issues/51540) or
 [#66741](https://github.com/rust-lang/rust/issues/66741) for tracking issues).
```
[the remark this answers is not in the document]
Comment (ogoffart) [reply api/rs/slint/mcu.md:20]: This comments on a oudated version of the wording.
I think the current wording is good enough.
If you think it could be improved, feel free to change it.
